aspnetcore/signalr/streaming/sample.netcoreapp3.0/Hubs/StreamHub.cs

```diff
@@ -0,0 +1,72 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Reactive.Linq;
+using System.Threading;
+using System.Threading.Channels;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.SignalR;
+
+namespace SignalRChat.Hubs
+{
+    public class StreamHub : Hub
+    {
+        #region snippet1
+        public ChannelReader<int> Counter(
+            int count,
+            int delay,
+            CancellationToken cancellationToken)
+        {
+            var channel = Channel.CreateUnbounded<int>();
+
+            // We don't want to await WriteItemsAsync, otherwise we'd end up waiting 
+            // for all the items to be written before returning the channel back to
+            // the client.
+            _ = WriteItemsAsync(channel.Writer, count, delay, cancellationToken);
+
+            return channel.Reader;
+        }
+
+        private async Task WriteItemsAsync(
+            ChannelWriter<int> writer,
+            int count,
+            int delay,
+            CancellationToken cancellationToken)
+        {
+            try
+            {
+                for (var i = 0; i < count; i++)
+                {
+                    // Check the cancellation token regularly so that the server will stop
+                    // producing items if the client disconnects.
+                    cancellationToken.ThrowIfCancellationRequested();
+                    await writer.WriteAsync(i);
```

Comment:
Pass the cancellationToken to ChannelWriter.WriteAsync. I don't think that the ThrowIfCancellationRequested is really necessary.